Clarify and correct Android Sections in WORKSPACE

jubjamie opened the pull request:
Build Tools version updated to latest version shipping with Android Studio.
Added some comments to clarify both this and NDK Level details that have been causing confusion.
Both of these have caused problems within Android studio so these edits should avoid the build tools error.

jubjamie committed:
Clarify and correct Android Sections

Build Tools version updated to latest version shipping with Android Studio.
Added some comments to clarify both this and NDK Level details that have been causing confusion.

tensorflow-jenkins commented:
Can one of the admins verify this patch?

jubjamie commented:
I signed it!

jubjamie committed:
Update Build Tools Version

Android studio uses 25.0.2 now (I think) and complains about incorrect bulid tools version in gradle. This fixes it. Not sure if this is unique to my android studio though because I have latest version?

jubjamie commented:
Added commit for build.gradle to match the build tools version change.

petewarden commented:
Andrew, can you take a look at this? Thanks!

andrewharp commented on WORKSPACE:
```diff
@@ -29,6 +30,7 @@ load("//tensorflow:workspace.bzl", "tf_workspace")
 #    name="androidndk",
 #    path="<PATH_TO_NDK>",
 #    api_level=14)
+#Note that API level is not the same as NDK version.
```
Can you sink to the latest version please and move this line above the newly added NDK comment? It's also missing a space after the #

jubjamie replied:
Sync to the latest version of what sorry? Tf? 1.1?
I will look at your comments on Monday and I'll get back to it then. I can also make similar edits to the previously mentioned files too if you want?

andrewharp replied:
Yes, TF master. I made some edits to this file internally to add comments here as well, and those changes have been pushed since you made your edits.

jubjamie replied:
I have made those changes.

andrewharp commented on WORKSPACE:
```diff
@@ -20,7 +20,8 @@ load("//tensorflow:workspace.bzl", "tf_workspace")
 #    api_level = 23,
-#    build_tools_version = "25.0.1",
+#    build_tools_version = "25.0.2",
+#    #Check your build tools version. Android Studio is shipping with 25.0.2 currently.
```
Is this correct? I see 25.0.4 in Android Studio now.

In any case I would:
1 ) move the comment above the build_tools_version line
2) rephrase it to say "# Ensure that you have a matching version of build_tools installed, as the SDK default changes periodically."

jubjamie replied:
After updating I only see 25.0.2 in AS within SDK manager and within my files on this ubuntu version of AS. I like your rephrasing as it allows for those differences.

andrewharp reviewed (changes requested):
The build_tools_version also need to be updated in 
tensorflow/tools/ci_build/Dockerfile.android and tensorflow/contrib/android/cmake/build.gradle, thanks

jubjamie committed:
Merge branch 'master' into master

jubjamie committed:
Requested changes to WORKSPACE for Android

Clarification of comments to the android section at the top. Comments added to avoid common confusion.

jubjamie committed:
Update Dockerfile.android to clarify build tools.

I don't understand this dockerfile too much but I gather that this will download and use the correct build_tools_version regardless? Nonetheless, updated to 25.0.2 as a minimum and added a comment to assist troubleshooters in the future.

jubjamie committed:
Update build_tools_version and add comment

Added comment to clarify build_tools_version updates and changed it to 25.0.2

jubjamie commented:
@andrewharp Those changes have been made however I am wondering whether it will break people's production environments? As a new starter with a fresh copy of android studio, I have one version of build_tools but others may still be on a previous version and this change may cause issues? This works both ways though as otherwise it breaks fresh builds on fresh android studio version. Although, curiously, you see a different build_tools_version to me and I've tried to update it! Thoughts?

andrewharp commented on WORKSPACE:
```diff
@@ -29,7 +30,9 @@ load("//tensorflow:workspace.bzl", "tf_workspace")
 #    name="androidndk",
 #    path="<PATH_TO_NDK>",
-#    api_level=14) # This needs to be 14 or higher to compile TensorFlow.
+#    # This needs to be 14 or higher to compile TensorFlow. Note NDK version is not the API level.
```
s/Note/Note that/

andrewharp commented on WORKSPACE:
```diff
@@ -20,7 +20,8 @@ load("//tensorflow:workspace.bzl", "tf_workspace")
 #    name = "androidsdk",
 #    api_level = 23,
-#    build_tools_version = "25.0.1",
+#    #Ensure that your build_tools_version (below) the SDK manager version as it updates periodicaly.
```
\# Ensure that you have the build_tools_version below installed in the SDK manager as it updates periodically.

andrewharp commented on tensorflow/tools/ci_build/Dockerfile.android:
```diff
@@ -29,7 +29,8 @@ RUN mkdir -p ${ANDROID_DEV_HOME}
 ENV ANDROID_SDK_URL https://dl.google.com/android/repository/${ANDROID_SDK_FILENAME}
 ENV ANDROID_API_LEVEL 23
-ENV ANDROID_BUILD_TOOLS_VERSION 25.0.1
+#Build Tools Version liable to change.
```
missing space after #

jubjamie committed:
Update WORKSPACE

jubjamie committed:
Update WORKSPACE

jubjamie committed:
Update Dockerfile.android

jubjamie reviewed (commented):
Changes reviewed and implemented.

andrewharp commented on WORKSPACE:
```diff
@@ -30,7 +30,7 @@ load("//tensorflow:workspace.bzl", "tf_workspace")
 #    name="androidndk",
 #    path="<PATH_TO_NDK>",
-#    # This needs to be 14 or higher to compile TensorFlow. Note NDK version is not the API level.
+#    # This needs to be 14 or higher to compile TensorFlow. Note that NDK version is not the API level.
```
Sorry, this should be "Note that the NDK version is not the API level."

andrewharp commented on WORKSPACE:
```diff
@@ -20,7 +20,8 @@ load("//tensorflow:workspace.bzl", "tf_workspace")
 #    name = "androidsdk",
 #    api_level = 23,
-#    build_tools_version = "25.0.1",
+#    # Ensure that you have the build_tools_version below installed in the SDK manager as it updates periodically.
```
line wrap at 80 characters

jubjamie committed:
Update WORKSPACE

jubjamie reviewed (commented):
Line Wrap implemented

andrewharp commented on WORKSPACE:
```diff
@@ -29,7 +31,9 @@ load("//tensorflow:workspace.bzl", "tf_workspace")
-#    api_level=14) # This needs to be 14 or higher to compile TensorFlow.
+#    # This needs to be 14 or higher to compile TensorFlow. Note that NDK version is not the API level.
+#    api_level=14)
+
```
extra blank line and missing "the"

jubjamie committed:
Updated for "the", lines and line wrapping

jubjamie reviewed (commented):
Requested changes made + new line for line wrapping.

yifeif commented:
@tensorflow-jenkins test this please

jubjamie commented:
Do I need to do anything to address the failed builds? I'm not sure they are failing and, considering how basic the changes were i'm not sure if there's anything I could do!
Thanks

yifeif commented:
@jubjamie thanks for asking. The failure looks like a flake. Running @tensorflow-jenkins test this please.

jubjamie commented:
@yifeif Hmmm ok i'll let Jenkins do his thing! Thanks

yifeif commented:
@tensorflow-jenkins test this please.
